internal/runbits/legacy/projectmigration/projectmigration.go

```diff
@@ -1,185 +1,84 @@
+	"github.com/go-openapi/strfmt"
 )
 
-//go:embed migrate.mac.bash
-var migrateMacScript []byte
-
-//go:embed migrate.nix.bash
-var migrateNixScript []byte
-
-//go:embed migrate.win.ps
-var migrateWinScript []byte
-
 type projecter interface {
-	Source() *projectfile.Project
 	Dir() string
-	URL() string
-	Path() string
 	LegacyCommitID() string
-	StripLegacyCommitID() error
+	SetLegacyCommit(commitID string) error
 }
 
-var prompter prompt.Prompter
-var out output.Outputer
-
-// Register exists to avoid boilerplate in passing prompt and out to every caller of
-// commitmediator.Get() for retrieving legacy commitId from activestate.yaml.
-// This is an anti-pattern and is only used to make this legacy feature palatable.
-func Register(prompter_ prompt.Prompter, out_ output.Outputer) {
-	prompter = prompter_
-	out = out_
+type migrator struct {
+	out  output.Outputer
+	proj projecter
 }
 
-func PromptAndMigrate(proj projecter) error {
-	if prompter == nil || out == nil {
-		return errs.New("projectmigration.Register() has not been called")
-	}
+func New(out output.Outputer, pj projecter) *migrator {
+	return &migrator{out: out, proj: pj}
+}
 
-	if os.Getenv(constants.DisableProjectMigrationPrompt) == "true" {
+// setupProject will ensure that the stored project matches the path that was requested. In most cases this should match,
+// and setting up a new project instance will be rare.
+func (m *migrator) setupProject(pjpath string) error {
+	if !ptr.IsNil(m.proj) && m.proj.Dir() == pjpath {
 		return nil
 	}
-
-	// We always set the local commit, the migration only touches on what happens with the commit in the activestate.yaml
-	if err := localcommit.Set(proj.Dir(), proj.LegacyCommitID()); err != nil {
-		return errs.Wrap(err, "Could not create local commit file")
-	}
-	for dir := proj.Dir(); filepath.Dir(dir) != dir; dir = filepath.Dir(dir) {
-		if !fileutils.DirExists(filepath.Join(dir, ".git")) {
-			continue
-		}
-		err := localcommit.AddToGitIgnore(dir)
-		if err != nil {
-			if !errors.Is(err, fs.ErrPermission) {
-				multilog.Error("Unable to add local commit file to .gitignore: %v", err)
-			}
-			out.Notice(locale.T("notice_commit_id_gitignore"))
-		}
-		break
-	}
-
-	// Prevent also showing the warning when we already prompt
-	warned = true
-
-	defaultChoice := false
-	if migrate, err := prompter.Confirm("", locale.T("projectmigration_confirm"), &defaultChoice); err == nil && !migrate {
-		if out.Config().Interactive {
-			out.Notice(locale.T("projectmigration_declined"))
-		}
-		return CreateMigrateScript(proj)
-	} else if err != nil {
-		return errs.Wrap(err, "Could not confirm migration choice")
-	}
-
-	if err := proj.StripLegacyCommitID(); err != nil {
-		return errs.Wrap(err, "Could not strip legacy commit ID")
-	}
-
-	out.Notice(locale.Tl("projectmigration_success", "Your project was successfully migrated"))
-
-	return nil
+	var err error
+	m.proj, err = project.FromPath(pjpath)
+	return err
```

Comment:
Do we want to wrap the error for stacktrace purposes? Or is it obvious enough where/how to track it down from a message/traceback?

Reply:
Yeah we should. I was being a bit lazy cause this is for handling legacy code, but you were right to flag me on it. 🙏